server/src/streaming/partitions/messages.rs

```diff
@@ -37,38 +37,35 @@ impl Partition {
         let mut found_index = None;
         let mut start_offset = self.segments.last().unwrap().start_offset;
         // Since index cache is configurable globally, not per segment we can handle it this way.
-        let cache_time_index = self.config.segment.cache_time_indexes;
-        if !cache_time_index {
+        if !self.config.segment.cache_indexes {
```

Comment:
this code is part of the `get_messages_by_timestamp` method. It used to lookup offset based on timestamp, and proxy it to `get_messages_by_offset` method. Since we store timestamp + position in our index, I think we can use that fact to call some lower level api (from storage) that fetches messages from given position.